platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemBuilder.java

```diff
@@ -225,19 +225,27 @@ public InternalProblemBuilder documentedAt(@Nullable DocLink doc) {
     }
 
     @Override
-    public InternalProblemBuilder id(String name, String displayName) {
-        this.id = new DefaultProblemId(name, displayName, cloneGroup(SharedProblemGroup.generic()));
+    public InternalProblemBuilder id(ProblemId problemId) {
+        if (problemId instanceof DefaultProblemId) {
+            this.id = problemId;
```

Comment:
💄 space

Reply:
I'm not sure what needs to  be adjusted here. I assume you mean the missing space on line 232.

Reply:
> https://github.com/gradle/gradle/pull/31778

Reply:
> missing space on line 232

is fixed.